src/steps/object-update.ts

```diff
@@ -0,0 +1,54 @@
+import { BaseStep, Field, StepInterface, ExpectedRecord } from '../core/base-step';
+import { Step, RunStepResponse, FieldDefinition, StepDefinition, RecordDefinition } from '../proto/cog_pb';
+
+export class UpdateObject extends BaseStep implements StepInterface {
+
+  protected stepName: string = 'Update a Salesforce Object';
+  protected stepExpression: string = 'update a salesforce (?<objName>[a-zA-Z0-9]+) object';
+  protected stepType: StepDefinition.Type = StepDefinition.Type.ACTION;
+  protected expectedFields: Field[] = [{
+    field: 'objName',
+    type: FieldDefinition.Type.STRING,
+    description: 'Salesforce object name',
+  }, {
+    field: 'salesforceObject',
+    type: FieldDefinition.Type.MAP,
+    description: 'where keys represent object field names as represented in the SFDC API',
+  }];
+  protected expectedRecords: ExpectedRecord[] = [{
+    id: 'salesforceObject ',
+    type: RecordDefinition.Type.KEYVALUE,
+    fields: [{
+      field: 'Id',
+      type: FieldDefinition.Type.STRING,
+      description: "Object's SalesForce ID",
+    }, {
+      field: 'CreatedDate',
+      type: FieldDefinition.Type.DATETIME,
+      description: "Object's Created Date",
+    }, {
+      field: 'LastModifiedDate',
+      type: FieldDefinition.Type.DATETIME,
+      description: "Object's Last Modified Date",
+    }],
+    dynamicFields: true,
+  }];
+
+  async executeStep(step: Step): Promise<RunStepResponse> {
+    const stepData: any = step.getData().toJavaScript();
+    const objName: any = stepData.objName;
+    const salesforceObject: any = stepData.salesforceObject;
+
+    try {
+      const result = await this.client.updateObject(objName, salesforceObject);
+      const record = this.keyValue('account', 'Created Object', { Id: result.id });
+
+      return this.pass('Successfully created Object with ID %s', [result.id], [record]);
+    } catch (e) {
+      return this.error('There was a problem creating the Object: %s', [e.toString()]);
```

Comment:
- [ ] Should be `There was a problem updating...`
- [ ] Same as above: consider adding the `objName` to make the message friendlier.